packages/node_modules/@webex/internal-plugin-wdm/src/device/device.js

```diff
@@ -524,29 +569,88 @@ const Device = WebexPlugin.extend({
+   * -1 which is the value set for duration option None/default value for the duration.
+   * It also checks if the url was reachable once then sets the isInNetwork to true there by preventing the reset timer
+   * from being called each time meeting events resets the timer.
+   */
+    if (!this.isInMeeting) {
+      if (this.config.enableInactivityEnforcement && this.intranetInactivityCheckUrl &&
+      this.intranetInactivityDuration && !this.isInNetwork && this.intranetInactivityDuration !== -1) {
+        this.on('change:lastUserActivityDate', () => { this._resetLogoutTimer(); });
+        const timer = safeSetTimeout(() => {
+          this.webex.request({
+            headers: {
+              'cisco-no-http-redirect': null,
+              'spark-user-agent': null,
+              trackingid: null
+            },
+            method: 'GET',
+            uri: this.intranetInactivityCheckUrl
           })
-          .catch((reason) => {
-            this.logger.warn('device: logout failed', reason);
-          });
-      }, this.intranetInactivityDuration * 1000);
+            .then(() => {
+              this.isInNetwork = true;
+            })
+            .catch(() => {
+              this.logger.info('device: did not reach internal ping endpoint; logging out after inactivity on a public network');
+
+              return this.webex.logout();
+            })
+            .catch((reason) => {
+              this.logger.warn('device: logout failed', reason);
+            });
+        }, this.intranetInactivityDuration * 1000);
+
+        this.logoutTimer = timer;
+      }
+    }
+  },
 
-      this.logoutTimer = timer;
+  _inNetworkResetLogoutTimer() {
+    clearTimeout(this.inNetworkLogoutTimer);
+    this.off('change:lastInNetworkUserActivityDate'); // removes previous event listener
+    this.unset('inNetworkLogoutTimer');
+    /**
+   * _inNetworkResetLogoutTimer
+   * Initially clears the timer and removes the onChange event listener for lastInNetworkUserActivityDate and unsets the timer
+   * Checks if the client is not in meeting
+   * Checks for the inactivityEnforcement, intranetInactivityCheckUrl, inNetworkInactivityDuration, also if
+   * inNetworkInactivityDuration isn't -1 which is the value set for duration option None/default value for the duration.
+   * It logs the user out on successful/unsuccessful ping of the url. This action is with the following point in mind,
+   * In-Network Duration is expected to be greater than the Off-Network so if the url turns out to be un-reachable its
+   * saft to log the user out.
+   */
+    if (!this.isInMeeting) {
+      if (this.config.enableInactivityEnforcement && this.intranetInactivityCheckUrl &&
+      this.inNetworkInactivityDuration && this.inNetworkInactivityDuration !== -1) {
+        this.on('change:lastInNetworkUserActivityDate', () => { this._inNetworkResetLogoutTimer(); });
+        const inNetworkTimer = safeSetTimeout(() => {
+          this.webex.request({
```

Comment:
If you are in network, why do we need to ping the same url that we are pinging to check if you are out of network? My main issue is it feels like we are duplicating a lot (almost all?) of the code paths? Can we not make do with one timer?

Reply:
The URL remains the same an internal network URL that can be reached only when connected to their network. When off network the off network timeout would log the user out if its not reachable. And while in network when the timer times out the user is logged out when the URL is reachable.

Reply:
What I'm saying is that no matter the result you do the same thing, log the user out. Why do you need to make the API call then if you don't care about the result?